internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -248,7 +255,92 @@ func (ctrl *ManagerController) Run(ctx context.Context, r controller.Runtime, lo
 			return fmt.Errorf("error creating address spec: %w", err)
 		}
 
-		// all done, terminate controller
-		return nil
+		return ctrl.pollConnection(ctx, logger)
```

Comment:
Note: this logic will again change on the implementation of this, which I'm already working on: https://github.com/siderolabs/talos/issues/7143

Reply:
I posted below, but my point it's easier to get "right" structure now